src/handlers/records-delete.ts

```diff
@@ -86,12 +90,15 @@ export class RecordsDeleteHandler implements MethodHandler {
     }
 
     const recordsWrite = await RecordsWrite.getInitialWrite(existingMessages);
-    const indexes = RecordsDeleteHandler.constructIndexes(recordsDelete, recordsWrite);
-    await this.messageStore.put(tenant, message, indexes);
-
+    const published = (newestExistingMessage as RecordsWriteMessage).descriptor.published;
+    const indexes = recordsDelete.constructIndexes(recordsWrite, published);
```

Comment:
Design: consider the following which I think shows the intent better.

```suggestion
    const indexes = recordsDelete.constructIndexes(recordsWrite, newestExistingMessage);
```